### PR TITLE
support updating tags for Topic

### PR DESCRIPTION
**Pull request by jaypipes**
Adds support for tagging and untagging Topic resources via the update code path.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by A-Hilaly — APPROVED** [review]
:+1:

File: pkg/util/tags.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	return tags
+}
+
+func equalStrings(a, b *string) bool {
+	if a == nil {
+		return b == nil || *b == ""
+	}
+	return (*a == "" && b == nil) || *a == *b
+}
```
[the remark this answers is not in the document]
Comment (jaypipes) [reply pkg/util/tags.go:79]: this was just copied from the IAM controller... should likely be placed into a `aws-controllers-k8s/pkg/` package at some point and reused.

[the remark this answers is not in the document]
Comment (A-Hilaly) [reply pkg/util/tags.go:79]: This is what I wanted to tackle with https://github.com/aws-controllers-k8s/pkg/pull/14 - should get back to it soon

**Pull request merged**